Documentation/learning/why.md

```diff
@@ -0,0 +1,6 @@
+# Why etcd
+
+etcd is designed to store a small amount of metadata in consistent, fault-tolerant way. Distributed systems use etcd as the root of scheduling and service discovery configuration storage. [Distributed lock][etcd-concurrency] is implemented on top of etcd, useful for master election in other distributed databases.
```

Comment:
I agree with xiang90. i think readme should be a short introduction about etcd. and why etcd should go in depth on what's etcd, its benefits over other solutions, and etc.